p2p/host/resource-manager/obs/stats.go

```diff
@@ -148,57 +198,61 @@ func (r StatsTraceReporter) ConsumeEvent(evt rcmgr.TraceEvt) {
 			peerStreamsOut := int64(evt.StreamsOut)
 			if oldStreamsOut != peerStreamsOut {
 				if oldStreamsOut != 0 {
-					stats.RecordWithTags(ctx, []tag.Mutator{tag.Upsert(directionTag, "outbound")}, peerStreamsNegative.M(oldStreamsOut))
+					previousPeerStreamsOutbound.Observe(float64(oldStreamsOut))
 				}
 				if peerStreamsOut != 0 {
-					stats.RecordWithTags(ctx, []tag.Mutator{tag.Upsert(directionTag, "outbound")}, peerStreams.M(peerStreamsOut))
+					peerStreamsOutbound.Observe(float64(peerStreamsOut))
 				}
 			}
 
 			oldStreamsIn := int64(evt.StreamsIn - evt.DeltaIn)
 			peerStreamsIn := int64(evt.StreamsIn)
 			if oldStreamsIn != peerStreamsIn {
 				if oldStreamsIn != 0 {
-					stats.RecordWithTags(ctx, []tag.Mutator{tag.Upsert(directionTag, "inbound")}, peerStreamsNegative.M(oldStreamsIn))
+					previousPeerStreamsInbound.Observe(float64(oldStreamsIn))
 				}
 				if peerStreamsIn != 0 {
-					stats.RecordWithTags(ctx, []tag.Mutator{tag.Upsert(directionTag, "inbound")}, peerStreams.M(peerStreamsIn))
+					peerStreamsInbound.Observe(float64(peerStreamsIn))
 				}
 			}
 		} else {
-			var tags []tag.Mutator
-			if rcmgr.IsSystemScope(evt.Name) || rcmgr.IsTransientScope(evt.Name) {
-				tags = append(tags, tag.Upsert(scopeTag, evt.Name))
-			} else if svc := rcmgr.ParseServiceScopeName(evt.Name); svc != "" {
-				tags = append(tags, tag.Upsert(scopeTag, "service"), tag.Upsert(serviceTag, svc))
-			} else if proto := rcmgr.ParseProtocolScopeName(evt.Name); proto != "" {
-				tags = append(tags, tag.Upsert(scopeTag, "protocol"), tag.Upsert(protocolTag, proto))
-			} else {
-				// Not measuring connscope, servicepeer and protocolpeer. Lots of data, and
-				// you can use aggregated peer stats + service stats to infer
-				// this.
-				break
-			}
-
 			if evt.DeltaOut != 0 {
-				stats.RecordWithTags(
-					ctx,
-					append([]tag.Mutator{tag.Upsert(directionTag, "outbound")}, tags...),
-					streams.M(int64(evt.DeltaOut)),
-				)
+				if rcmgr.IsSystemScope(evt.Name) || rcmgr.IsTransientScope(evt.Name) {
+					*tags = (*tags)[:0]
```

Comment:
Is this needed? `getStringSlice` already does that.

Reply:
Not needed in this case, but if the tags is reused elsewhere before the slice is returned it is (e.g. L238). Better to be consistent.